src/com/content-notifications/base.js

```diff
@@ -328,4 +339,46 @@ export default class NotificationsBase extends Component {
 		return Notifications;
 	}
 
+	shouldShowNotification(notification) {
+		const {mention, friendGame, friendPost, feedback, comment, other} = this.state.filters;
+		if (feedback !== false && isNotificationFeedback(notification)) {
+			return true;
+		}
+		else if (mention !== false && isNotificationMention(notification)) {
+			return true;
+		}
+		else if (friendGame !== false && isNotificationFriendGame(notification)) {
+			return true;
+		}
+		else if (friendPost !== false && isNotificationFriendPost(notification)) {
+			return true;
+		}
+		else if (comment !== false && isNotifactionComment(notification)) {
+			return true;
+		}
+		return other !== false && isNotificationOther(notification);
+	}
+
+	handleFilterChange(filterType, otherStuff) {
+		let oldFilter = Object.assign({}, this.state.filters);
+		if (filterType == 'comment') {
```

Comment:
I can't help but notice all these if statements could be replaced by a single line using oldFilter[filterType] = ...

Reply:
That was my original idea, but I think the linter got really upset about it. I can give it a try and see if I'm remembering wrong.

Reply:
Feel free to share the line you want to make linter friendly. We can help.